src/index.ts

```diff
@@ -37,10 +41,157 @@ function gistIdFromTwiddleUrl(twiddleUrl: string): string {
   }
 }
 
-export default function twiddleEject(twiddleUrl: string, projectName: string, _options: {} = {}) {
+/**
+ * Given a path, generate single or multi level directory.
+ * @param path
+ */
+function generateDir(path: string) {
+  const fullPath = path.split('/').reduce((accumulator, current) => {
+    if (!fs.existsSync(accumulator)) {
+      fs.mkdirSync(accumulator);
+    }
+    return `${accumulator}/${current}`;
+  });
+  if (!fs.existsSync(fullPath)) {
+    fs.mkdirSync(fullPath);
+  }
+}
+
+/**
+ * Unzip and extract from a given url and save the data to a path provided.
+ * @param gistUrl
+ * @param filePath
+ * @param targetDir
+ */
+async function unzipAndExtractFromUrl(gistUrl: string, filePath: string, targetDir: string): Promise<string> {
+  const gistLink = await fetch(gistUrl).then(res => res.url);
```

Comment:
Avoid mixing async/await with traditional promise handling (i.e., `.then`)
```suggestion
  const { url: gistLink } = await fetch(gistUrl)
```